Source/player.cpp

```diff
@@ -1055,10 +1052,10 @@ bool DoAttack(Player &player)
 			}
 		}
 
-		if (dMonster[dx][dy] != 0) {
-			didhit = PlrHitMonst(player, abs(dMonster[dx][dy]) - 1);
-		} else if (dPlayer[dx][dy] != 0 && !player.friendlyMode) {
-			didhit = PlrHitPlr(player, Players[abs(dPlayer[dx][dy]) - 1]);
+		if (monster != nullptr) {
+			didhit = PlrHitMonst(player, *monster);
+		} else if (PlayerAtPosition(position) != nullptr && !player.friendlyMode) {

```

Comment:
There's no nice way to structure this to avoid calling `PlayerAtPosition` twice(or doing it unnecessarily before the if statements :/). Would be worth reordering the conditions so the PVP flag is checked first but I'd do that in a separate PR and include things like the calls to `CanTalkToMonst` in the cleave section.

Reply:
There's the "If statement with initializer" from C++ 17 but I heard that that feature is not very popular 😁

Reply:
yeah, I'd rate that a "not-nice" way of doing it :D